src/platform/i386/kernel.c

```diff
@@ -91,15 +91,20 @@ kern_memory_setup(struct multiboot *mb, u32_t mboot_magic)
 		struct multiboot_mem_list *mem      = &mems[i];
 		u8_t *                     mod_end  = glb_memlayout.mod_end;
 		u8_t *                     mem_addr = chal_pa2va((paddr_t)mem->addr);
+		unsigned long              mem_len  = (mem->len > COS_PHYMEM_MAX_SZ ? COS_PHYMEM_MAX_SZ : mem->len); /* maximum allowed */ 
 
-		printk("\t- %d (%s): [%08llx, %08llx)\n", i, mem->type == 1 ? "Available" : "Reserved ", mem->addr,
-		       mem->addr + mem->len);
+		if (mem->addr > BOOT_KERN_MEMSCAN_MAX || mem->addr + mem_len > BOOT_KERN_MEMSCAN_MAX) {
+			printk("\tScanned maximum allowed regions\n");
+			break;
```

Comment:
I forget:  does multiboot guarantee that the memory maps are ordered from lower to higher addresses?  If not, we don't want to break out, right?  Just `continue`?

Do you know?

Reply:
No I don't know the specification for that, will look into it and fix it to use `continue` if there is no specification on the order.